include/nrf_rpc/nrf_rpc_uart.h

```diff
@@ -0,0 +1,82 @@
+/*
```

Comment:
Is it really necessary to new create types over `typedef` rather than just using `struct nrf_rpc_uart` and `enum nrf_rpc_hdlc_state`? Seems usage of structure and enumeration names matches Zephyr's paradigm better.